src/modules/auth/auth.emailVerificationService.ts

```diff
@@ -0,0 +1,56 @@
+import * as nodemailer from 'nodemailer';
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { User } from '@/modules/user/models/user.entity';
+import { Result } from '@/common/dto/result.type';
+import { REGISTER_ERROR } from '@/common/constants/code';
+
+@Injectable()
+export class EmailVerificationService {
+  constructor(private jwtService: JwtService) {}
+
+  generateVerificationToken(email: string): string {
+    const token = this.jwtService.sign({ email }, { expiresIn: '30m' });
+    return token;
+  }
+
+  async sendVerificationEmail(user: User): Promise<Result> {
+    const transporter = nodemailer.createTransport({
+      service: 'Gmail',
+      host: 'smtp.gmail.com',
+      port: 465,
+      secure: true,
+      auth: {
+        user: process.env.EMAIL_USER,
+        pass: process.env.EMAIL_PASS,
+      },
+    });
+
+    const verificationLink = `http://localhost:5173/verify-email?email=${user.email}&token=${user.verificationToken}`;
```

Comment:
for this temp url, needs to add `TODO: link will be updated once api server is deployed, currently this url is for local development only`